mantidproject/mantid
# Fix a couple property handler bugs

### Description of work

Initially this work was just to fix #37129, but I also found a bug where all the remaining "Fix" properties would be removed when removing a function from the same composite function.

#### Summary of work
- To fix the original issue, the parameter index needed to be retrieved from the top level composite function so the long prefix (`f0.f0.f0`) could be parsed properly.
- To fix the issue with removing the Fix sub properties, a check has been added to `renameChildren` to see if the parameter is fixed.
- I noticed that when adding the first function to a convolution, it sets all parameters to be fixed. When this gets initialised in the property handler, I've changed it to use the `fix` function so it is all more consistent
- I've also reduced some code reuse by changing `fix` to call `addTie` which now checks if the tie is static (aka a fix).

Fixes #37129

`renameChildren` definitely still contains a problem or two. I don't think it copes well if you had a bunch of nested composite function and wanted to tie across different levels, but this isn't a regression, so I will try and fix these in a later pr.

### To test:

 - Follow the instructions from the issue
 - Clear the model
 - Add two Gaussians
 - Right click to fix the height of one of the Gaussians
 - Remove the other Gaussian, the fix property should remain in place.



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
